### DISCUSSION TITLE
How to use CSS Grid in DaisyUI / Feliz / SafeStack project

### DISCUSSION THREAD

**Discussion by WarmWinters** [Zaid-Ajaj/Feliz#493] [Q&A] [answered]
What is the correct way to use CSS Grid in a SafeStack / Feliz.DaisyUI project?   

I have this basic layout (code below) in html / CSS and want to mirror the structure in index.fs Html.body [ ] with DaisyUI.    Feliz.DaisyUI or Felix.Bulma components are working in my app but I am stuck on the correct way to use CSS grid for the layout. So far have not found any examples or figured out the code.   The app was created with the Safestack template and updated to .net60.  Many Thanks. 

index.html
```
 <!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link rel="stylesheet" href="grid2.css" />
    <title>CSS Grid Basic</title>
</head>
<body>
    <header>Header</header>
    <main>Content</main>
    <nav>Navigation</nav> 
    <footer>Footer</footer>
</body>
</html>
```
index.css
```
* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

body {
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  height: 100vh;

  display: grid;
  grid-template-areas:
    'header header header'
    'nav content content'
    'footer footer footer';
  grid-template-columns: 1fr 4fr 1fr;
  grid-template-rows: 80px 1fr 50px;
}

header,
footer,
main,
nav {
  background-color: steelblue;
  color: #fff;
  padding: 20px;
  border: skyblue 1px solid;
}
header {
  grid-area: header;
}
nav {
  grid-area: nav;
}
main {
  grid-area: content;
}
footer {
  grid-area: footer;
}
```

**Comment by MangelMaxime** [comment, marked as answer]
Hello @WarmWinters,

I am not sure to understand what you are trying to do.

It seems like your `index.html` is containing some static HTML in the body where in general you just have a `root` div in the body and then use React to build the application from that.

Then using CSS classes is the same as in any React application, you import your style file in your application (it depends on the bundler you are using).

And then you can write things like:

```fs

Html.div [
    Html.header [
        prop.text "Header"
    ]
    Html.main [
        prop.text "Content"
    ]
    Html.nav [
        prop.text "Navigation"
    ]
    Html.footer [
        prop.text "Footer"
    ]
]
```

In general, it is recommended to use CSS classes instead of writing the rule to apply to all the element of a given type.

Because, we what you wrote all the `footer` elements are going to be style the same way.

If you use CSS classes, you need to use `prop.className` to set it to the React element.

```fs
Html.footer [
    prop.className "my-footer-css-class"
    prop.text "Footer"
]
```

**Comment by WarmWinters** [comment]
Thanks for the detailed reply @MangelMaxime.   I was not clear how to use CSS grid with Feliz but I figured it out eventually and like everything else its simple when you know how.   The Feliz documentation and examples are excellent but it still takes a while to figure out how it all works.  I am really enjoying learning and using Feliz.
